src/ui/public/saved_objects/saved_object.js

```diff
@@ -0,0 +1,36 @@
+import _ from 'lodash';
+
+export class SavedObject {
+  constructor(client, attributes) {
+    this.client = client;
+    this.attributes = attributes;
+  }
+
+  get(key) {
+    return _.get(this.attributes, key);
+  }
+
+  set(key, value) {
+    return _.set(this.attributes, key, value);
+  }
+
+  save() {
+    if (this.id) {
+      return this.client.update(this.type, this.id, this.attributes);
+    } else {
+      return this.client.create(this.type, this.attributes);
+    }
+  }
+
+  delete() {
+    return this.client.delete(this.type, this.id);
+  }
+
+  get id() {
```

Comment:
Why have special getters for the id and type properties? Seems odd to have special cases for these two fields.

Maybe this is for backwards compatibility? If so I'd love if the first time these properties are accessed they logged a deprecation warning.